### PR TITLE
refactor: port server-variable model to ts

### PR DESCRIPTION
**Pull request by Souvikns**
**Description**

Porting `server-variable` object to typescript. 

**Related issue(s)**

See also #482

### CODE REVIEW

**Comment by Souvikns** [comment]
Is `enum` property in the [spec](https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md#server-variable-object) maps to the `allowedValues` in the new server-api?

@smoya @magicmatatjahu

File: src/models/v2/server.ts [hunk 1]
```diff
@@ -36,4 +39,15 @@ export class Server extends Mixin(BaseModel, BindingsMixin, DescriptionMixin, Ex
...
     return this._json.protocolVersion;
   }
+
+  variables(): ServerVariablesInterface {
+    return new ServerVariables(
+      Object.entries(
+        this._json.variables
+      ).map(
+        ([serverVariableName, serverVariable]) => this.createModel(
+          ServerVariable, serverVariable, { id: serverVariableName, pointer: `variables/${serverVariableName}` }
```
Comment (magicmatatjahu) [inline src/models/v2/server.ts:49]:
```suggestion
          ServerVariable, serverVariable, { id: serverVariableName, pointer: `${this._meta.pointer}/variables/${serverVariableName}` }
```

File: src/models/v3/server.ts [hunk 2]
```diff
@@ -36,4 +39,15 @@ export class Server extends Mixin(BaseModel, BindingsMixin, DescriptionMixin, Ex
...
     return this._json.protocolVersion;
   }
+
+  variables(): ServerVariablesInterface {
+    return new ServerVariables(
+      Object.entries(
+        this._json.variables
+      ).map(
+        ([serverVariableName, serverVariable]) => this.createModel(
+          ServerVariable, serverVariable, { id: serverVariableName, pointer: `variables/${serverVariableName}` }
```
Comment (magicmatatjahu) [inline src/models/v3/server.ts:49]:
```suggestion
          ServerVariable, serverVariable, { id: serverVariableName, pointer: `${this._meta.pointer}/variables/${serverVariableName}` }
```

**Review by magicmatatjahu — COMMENTED** [review]
Two improvements :) Overall good.

**Comment by Souvikns** [comment]
Well, there are 30 code smells, all `Unnecessary semicolon`.

**Review by magicmatatjahu — APPROVED** [review]
LGTM 🚀

**Comment by magicmatatjahu** [comment]
> Well, there are 30 code smells, all Unnecessary semicolon.

Atm it's not the problem :) We will resolve that issues with eslint.

**Pull request merged**